lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +232,97 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self._full_url = full_url
+        self._article_id = article_id
+        self._config = config
+        self.article = Article(self._full_url, self._article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        main_bs = article_soup.find('div', {'class': "page-content io-article-body"})
+        for paragraph in main_bs.find_all('p')[:-1]:
+            self.article.text += paragraph.text + '\n'
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        self.article.title = article_soup.find('title').text
+        author = article_soup.find('meta', {'name': "Author"})
+        if not author.text:
+            self.article.author = ['NOT FOUND']
+        else:
+            self.article.author = [author.text]
+        self.article.topics.append(article_soup.find('a', {'class': "fn-rubric-a"}).text)
+        date = article_soup.find('div', {'class': "fn-rubric-link"})
+        self.article.date = self.unify_date_format(date.text.strip())
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        pattern = '%d %m %Y, %H:%M'
+        today = datetime.datetime.now()
+        if len(date_str) == 5:
+            current_date = f'{today.day} {today.month} {today.year}, {date_str}'
+            return datetime.datetime.strptime(current_date, pattern)
+        months = {'января': '01',
+                  'февраля': '02',
+                  'марта': '03',
+                  'апреля': '04',
+                  'мая': '05',
+                  'июня': '06',
+                  'июля': '07',
+                  'августа': '08',
+                  'сентября': '09',
+                  'октября': '10',
+                  'ноября': '11',
+                  'декабря': '12'}
+        for month_name, month_number in months.items():
+            if month_name in date_str:
+                date_str = date_str.replace(month_name, month_number)
+                if f'{month_number},' in date_str:
+                    date_str = (date_str[:date_str.find(',')]
+                                + f' {today.year}'
+                                + date_str[date_str.find(','):])
+        return datetime.datetime.strptime(date_str, pattern)
```

Comment:
can you try to simplify date extraction?

Reply:
i think it is quite complicated now

Reply:
I guess 

> i think it is quite complicated now

I changed it.. I guess it's simpler now